### ISSUE TITLE
Couldn't join game in TShock Mintaka Unstable build 1386

### ISSUE THREAD

**Issue by mistzzt** [NyxStudios/TShock#1340]
Terraria would always be "Requesting world information".

But if you revert [this commit 1741b0c](https://github.com/NyxStudios/TShock/commit/1741b0c854311a23814731b5acfed6b72e9c75a2), players will be able to join the server again.

**Comment by mistzzt** [comment]
2016-12-12 18:57:22 - GetDataHandlers: ERROR: System.IndexOutOfRangeException: 索引超出了数组界限。
   在 TShockAPI.PlayerData.CopyCharacter(TSPlayer player)
   在 TShockAPI.GetDataHandlers.HandleConnecting(GetDataHandlerArgs args)
   在 TShockAPI.GetDataHandlers.HandlerGetData(PacketTypes type, TSPlayer player, MemoryStream data)

**Comment by ProfessorXZ** [comment]
Backup your database and delete it. See if the issue persists afterwards.

**Comment by ProfessorXZ** [comment]
One more thing I'd like to point out; reverting the commit you posted will mess up the inventory (issue #1337). I've also got confirmation the said commit fixes the issue.

**Comment by mistzzt** [comment]
The issue persists even after I deleted the database.

Actually, I also tried to delete all configuarations files, the problem still exists.

**Comment by Enerdy** [comment]
Reproduction steps?

**Comment by mistzzt** [comment]
just download the **Unstable build 1386** and join the server
![example](https://cloud.githubusercontent.com/assets/12794966/21103739/adef0730-c0be-11e6-8530-49068064fe07.png)

**Comment by MineBartekSA** [comment]
I have this too i just donwload lastest version of mintaka (1389) and i can't join.
In log i have this error : 
`GetDataHandlers: ERROR: System.IndexOutOfRangeException: Indeks wykraczał poza granice tablicy.
   w TShockAPI.PlayerData.CopyCharacter(TSPlayer player)
   w TShockAPI.GetDataHandlers.HandleConnecting(GetDataHandlerArgs args)
   w TShockAPI.GetDataHandlers.HandlerGetData(PacketTypes type, TSPlayer player, MemoryStream data)`

**Comment by nicatronTg** [comment]
Can confirm this on 1389. Noted that there's no "created unix socket" in the console.

```
Kurisu 福 ~/Downloads/tshock_debug (2)
3332 ◯ : mono --debug --trace=N:nothing TerrariaServer.exe
[0x7fffc28763c0:] EXCEPTION handling: System.ComponentModel.Win32Exception: Success
TerrariaAPI Version: 2.0.0.0 (Protocol v1.3.4.4 (188), OTAPI 1.3.4.4)
TShock 4.3.21.0 (Mintaka) now running.
AutoSave Enabled
Backups Disabled
Welcome to TShock for Terraria. Initialization complete.
[Server API] Info Plugin TShock v4.3.21.0 (by The TShock Team) initiated.
Terraria Server v1.3.4.4

1		Aleah
2		Aleah's Heart
3		Aleah's Heartbeats
n	New World
d <number>Delete World

Choose World: 3
Terraria Server v1.3.4.4

Max players (press enter for 8):
Terraria Server v1.3.4.4

Server port (press enter for 7777):
Terraria Server v1.3.4.4

Automatically forward port? (y/n):
Terraria Server v1.3.4.4

Server password (press enter for none):
Terraria Server v1.3.4.4

Resetting game objects 1%
Resetting game objects 2%
Resetting game objects 3%
Resetting game objects 4%
Resetting game objects 5%
Resetting game objects 6%
Resetting game objects 7%
Resetting game objects 8%
Resetting game objects 9%
Resetting game objects 10%
Resetting game objects 11%
Resetting game objects 12%
Resetting game objects 13%
Resetting game objects 14%
Resetting game objects 15%
Resetting game objects 16%
Resetting game objects 17%
Resetting game objects 18%
Resetting game objects 19%
Resetting game objects 20%
Resetting game objects 21%
Resetting game objects 22%
Resetting game objects 23%
Resetting game objects 24%
Resetting game objects 25%
Resetting game objects 26%
Resetting game objects 27%
Resetting game objects 28%
Resetting game objects 29%
Resetting game objects 30%
Resetting game objects 31%
Resetting game objects 32%
Resetting game objects 33%
Resetting game objects 34%
Resetting game objects 35%
Resetting game objects 36%
Resetting game objects 37%
Resetting game objects 38%
Resetting game objects 39%
Resetting game objects 40%
Resetting game objects 41%
Resetting game objects 42%
Resetting game objects 43%
Resetting game objects 44%
Resetting game objects 45%
Resetting game objects 46%
Resetting game objects 47%
Resetting game objects 48%
Resetting game objects 49%
Resetting game objects 50%
Resetting game objects 51%
Resetting game objects 52%
Resetting game objects 53%
Resetting game objects 54%
Resetting game objects 55%
Resetting game objects 56%
Resetting game objects 57%
Resetting game objects 58%
Resetting game objects 59%
Resetting game objects 60%
Resetting game objects 61%
Resetting game objects 62%
Resetting game objects 63%
Resetting game objects 64%
Resetting game objects 65%
Resetting game objects 66%
Resetting game objects 67%
Resetting game objects 68%
Resetting game objects 69%
Resetting game objects 70%
Resetting game objects 71%
Resetting game objects 72%
Resetting game objects 73%
Resetting game objects 74%
Resetting game objects 75%
Resetting game objects 76%
Resetting game objects 77%
Resetting game objects 78%
Resetting game objects 79%
Resetting game objects 80%
Resetting game objects 81%
Resetting game objects 82%
Resetting game objects 83%
Resetting game objects 84%
Resetting game objects 85%
Resetting game objects 86%
Resetting game objects 87%
Resetting game objects 88%
Resetting game objects 89%
Resetting game objects 90%
Resetting game objects 91%
Resetting game objects 92%
Resetting game objects 93%
Resetting game objects 94%
Resetting game objects 95%
Resetting game objects 96%
Resetting game objects 97%
Resetting game objects 98%
Resetting game objects 99%
Resetting game objects 100%
Loading world data: 1%
Loading world data: 2%
Loading world data: 3%
Loading world data: 4%
Loading world data: 5%
Loading world data: 6%
Loading world data: 7%
Loading world data: 8%
Loading world data: 9%
Loading world data: 10%
Loading world data: 11%
Loading world data: 12%
Loading world data: 13%
Loading world data: 14%
Loading world data: 15%
Loading world data: 16%
Loading world data: 17%
Loading world data: 18%
Loading world data: 19%
Loading world data: 20%
Loading world data: 21%
Loading world data: 22%
Loading world data: 23%
Loading world data: 24%
Loading world data: 25%
Loading world data: 26%
Loading world data: 27%
Loading world data: 28%
Loading world data: 29%
Loading world data: 30%
Loading world data: 31%
Loading world data: 32%
Loading world data: 33%
Loading world data: 34%
Loading world data: 35%
Loading world data: 36%
Loading world data: 37%
Loading world data: 38%
Loading world data: 39%
Loading world data: 40%
Loading world data: 41%
Loading world data: 42%
Loading world data: 43%
Loading world data: 44%
Loading world data: 45%
Loading world data: 46%
Loading world data: 47%
Loading world data: 48%
Loading world data: 49%
Loading world data: 50%
Loading world data: 51%
Loading world data: 52%
Loading world data: 53%
Loading world data: 54%
Loading world data: 55%
Loading world data: 56%
Loading world data: 57%
Loading world data: 58%
Loading world data: 59%
Loading world data: 60%
Loading world data: 61%
Loading world data: 62%
Loading world data: 63%
Loading world data: 64%
Loading world data: 65%
Loading world data: 66%
Loading world data: 67%
Loading world data: 68%
Loading world data: 69%
Loading world data: 70%
Loading world data: 71%
Loading world data: 72%
Loading world data: 73%
Loading world data: 74%
Loading world data: 75%
Loading world data: 76%
Loading world data: 77%
Loading world data: 78%
Loading world data: 79%
Loading world data: 80%
Loading world data: 81%
Loading world data: 82%
Loading world data: 83%
Loading world data: 84%
Loading world data: 85%
Loading world data: 86%
Loading world data: 87%
Loading world data: 88%
Loading world data: 89%
Loading world data: 90%
Loading world data: 91%
Loading world data: 92%
Loading world data: 93%
Loading world data: 94%
Loading world data: 95%
Loading world data: 96%
Loading world data: 97%
Loading world data: 98%
Loading world data: 99%
Loading world data: 100%
Settling liquids 1%
Settling liquids 2%
Settling liquids 3%
Settling liquids 4%
Settling liquids 5%
Settling liquids 6%
Settling liquids 7%
Settling liquids 8%
Settling liquids 9%
Settling liquids 10%
Settling liquids 11%
Settling liquids 12%
Settling liquids 13%
Settling liquids 14%
Settling liquids 15%
Settling liquids 16%
Settling liquids 17%
Settling liquids 18%
Settling liquids 19%
Settling liquids 20%
Settling liquids 21%
Settling liquids 22%
Settling liquids 23%
Settling liquids 24%
Settling liquids 25%
Settling liquids 26%
Settling liquids 27%
Settling liquids 28%
Settling liquids 29%
Settling liquids 30%
Settling liquids 31%
Settling liquids 32%
Settling liquids 33%
Settling liquids 34%
Settling liquids 35%
Settling liquids 36%
Settling liquids 37%
Settling liquids 38%
Settling liquids 39%
Settling liquids 40%
Settling liquids 41%
Settling liquids 42%
Settling liquids 43%
Settling liquids 44%
Settling liquids 45%
Settling liquids 46%
Settling liquids 47%
Settling liquids 48%
Settling liquids 49%
Settling liquids 50%
Settling liquids 93%
Settling liquids 97%
Settling liquids 98%
Starting server...
Server started
Terraria Server v1.3.4.4

Listening on port 7777
Type 'help' for a list of commands.

: TShock Notice: authcode.txt is still present, and the AuthToken located in that file will be used.
To become superadmin, join the game and type /auth 4930086
This token will display until disabled by verification. (/auth)
: 127.0.0.1:57163 is connecting...
127.0.0.1:57163 is connecting...
[0x70000679f000:] EXCEPTION handling: TShockAPI.DB.UserNotExistException: User 'Shark' does not exist
[0x70000679f000:] EXCEPTION handling: System.IndexOutOfRangeException: Index was outside the bounds of the array.
[0x700006ba5000:] EXCEPTION handling: System.Net.Sockets.SocketException: The socket has been shut down
[0x700006ba5000:] EXCEPTION handling: System.IO.IOException: Unable to write data to the transport connection: The socket has been shut down.
stop
Server executed: /stop.
Saving world. Momentary lag might result from this.
Saving world data: 1%
Saving world data: 3%
Saving world data: 4%
Saving world data: 5%
Saving world data: 6%
Saving world data: 8%
Saving world data: 9%
Saving world data: 10%
Saving world data: 11%
Saving world data: 12%
Saving world data: 13%
Saving world data: 15%
Saving world data: 16%
Saving world data: 17%
Saving world data: 18%
Saving world data: 19%
Saving world data: 20%
Saving world data: 21%
Saving world data: 22%
Saving world data: 23%
Saving world data: 24%
Saving world data: 26%
Saving world data: 28%
Saving world data: 29%
Saving world data: 30%
Saving world data: 31%
Saving world data: 32%
Saving world data: 33%
Saving world data: 34%
Saving world data: 35%
Saving world data: 36%
Saving world data: 37%
Saving world data: 38%
Saving world data: 40%
Saving world data: 41%
Saving world data: 42%
Saving world data: 43%
Saving world data: 44%
Saving world data: 45%
Saving world data: 46%
Saving world data: 47%
Saving world data: 48%
Saving world data: 49%
Saving world data: 51%
Saving world data: 52%
Saving world data: 53%
Saving world data: 54%
Saving world data: 55%
Saving world data: 56%
Saving world data: 57%
Saving world data: 59%
Saving world data: 60%
Saving world data: 61%
Saving world data: 62%
Saving world data: 63%
Saving world data: 64%
Saving world data: 65%
Saving world data: 66%
Saving world data: 67%
Saving world data: 68%
Saving world data: 69%
Saving world data: 70%
Saving world data: 71%
Saving world data: 72%
Saving world data: 73%
Saving world data: 74%
Saving world data: 75%
Saving world data: 76%
Saving world data: 77%
Saving world data: 78%
Saving world data: 79%
Saving world data: 80%
Saving world data: 81%
Saving world data: 82%
Saving world data: 83%
Saving world data: 84%
Saving world data: 85%
Saving world data: 86%
Saving world data: 88%
Saving world data: 90%
Saving world data: 91%
Saving world data: 92%
Saving world data: 93%
Saving world data: 94%
Saving world data: 95%
Saving world data: 96%
Saving world data: 97%
Saving world data: 99%
Saving world data: 100%
Validating world save: 14%
Validating world save: 25%
Validating world save: 40%
Validating world save: 47%
Validating world save: 61%
Validating world save: 67%
Validating world save: 79%
Validating world save: 88%
Backing up world file
World saved.
Server shutting down!
[0x700006da8000:] EXCEPTION handling: System.Threading.ThreadAbortException:
[0x700006fab000:] EXCEPTION handling: System.Threading.ThreadAbortException:
[0x7000071ae000:] EXCEPTION handling: System.Threading.ThreadAbortException:
```

```
2016-12-12 10:55:29 - TextLog: INFO: TShock 4.3.21.0 (Mintaka) now running.
2016-12-12 10:55:29 - TShock: INFO: AutoSave Enabled
2016-12-12 10:55:29 - TShock: INFO: Backups Disabled
2016-12-12 10:55:29 - TShock: INFO: Welcome to TShock for Terraria. Initialization complete.
2016-12-12 10:55:56 - GetDataHandlers: ERROR: System.IndexOutOfRangeException: Index was outside the bounds of the array.
  at TShockAPI.PlayerData.CopyCharacter (TShockAPI.TSPlayer player) [0x00513] in <f8375f72f3964138988795f053690626>:0
  at TShockAPI.GetDataHandlers.HandleConnecting (TShockAPI.GetDataHandlerArgs args) [0x0003e] in <f8375f72f3964138988795f053690626>:0
  at TShockAPI.GetDataHandlers.HandlerGetData (PacketTypes type, TShockAPI.TSPlayer player, System.IO.MemoryStream data) [0x0001d] in <f8375f72f3964138988795f053690626>:0
2016-12-12 10:56:05 - TextLog: INFO: Shark disconnected.
2016-12-12 10:56:12 - Utils: INFO: Server executed: /stop.
2016-12-12 10:56:12 - Utils: INFO: Broadcast: Saving world. Momentary lag might result from this.
2016-12-12 10:56:15 - Utils: INFO: Broadcast: World saved.
2016-12-12 10:56:15 - SaveManager: INFO: World saved at (/Users/shank/My Games/Terraria/Worlds/Aleah-s_Heartbeats.wld)
2016-12-12 10:56:15 - Utils: INFO: Broadcast: Server shutting down!
```

```
Kurisu 福 ~/Downloads/tshock_debug (2)/tshock
3339 ◯ : mono --version
Mono JIT compiler version 4.6.0 (Stable 4.6.0.245/746756c Mon Sep 19 13:57:55 BST 2016)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           normal
	SIGSEGV:       altstack
	Notification:  kqueue
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug
	LLVM:          supported, not enabled.
	GC:            sgen
```

**Comment by Exeden** [comment]
Not sure if related, but clean install, running Ubuntu-mono, and seems to hang at the below screen. Checked logs, etc, but not finding anything.
```
Terraria Server v1.3.4.4

Listening on port 7777
Type 'help' for a list of commands.

TShock Notice: To become SuperAdmin, join the game and type /auth 9813822
This token will display until disabled by verification. (/auth)

168.215.47.12:24577 is connecting...
168.215.47.12:24577 is connecting...
Creating linux socket
```
![2016-12-12_16-21-39](https://cloud.githubusercontent.com/assets/10290610/21117367/4173ecba-c087-11e6-9c2d-f07c3ebb4d7d.jpg)

**Comment by Enerdy** [comment]
The "stuck at Found Server" issues were likely fixed in #1342 seeing as everyone testing it was able to connect.
